userspace/libpman/src/ringbuffer.c

```diff
@@ -86,130 +93,153 @@ int pman_prepare_ringbuf_array_before_loading()
 	return err;
 }
 
-static int create_first_ringbuffer_map()
+static bool is_cpu_online(uint16_t cpu_id)
 {
-	int ringubuf_array_fd = -1;
-	int ringbuf_map_fd = -1;
-	int index = 0;
-
-	/* We don't need anymore the inner map, close it. */
-	close(g_state.inner_ringbuf_map_fd);
-
-	/* `ringbuf_array` is a maps array, every map inside it is a `BPF_MAP_TYPE_RINGBUF`. */
-	ringubuf_array_fd = bpf_map__fd(g_state.skel->maps.ringbuf_maps);
-	if(ringubuf_array_fd <= 0)
+	/* CPU 0 is always online */
+	if(cpu_id == 0)
```

Comment:
Hmm... I would says this is assumption is true on a single-CPU system. Otherwise, CPU 0 could be set off-line, e.g., `chcpu -d 0` 
```
CPU(s):                  4
  On-line CPU(s) list:   1,2
  Off-line CPU(s) list:  0,3
```

Reply:
Tha's true the real issue here is that `cpu0` has not the `/sys/devices/system/cpu/cpu0/online` file that we use here... same for old BPF https://github.com/falcosecurity/libs/blob/master/userspace/libscap/engine/bpf/scap_bpf.c#L1471

Reply:
might have changed these days:
````
# cat /sys/devices/system/cpu/cpu0/online 
1
````

Reply:
Mmmh cpu0/online does not exist for me either, on 
```
uname -r
6.1.4-arch1-1
```

Perhaps some cpus allow the cpu0 to be disabled and the kernel then exposes their `online` file too?

Reply:
Could be or it could be also related to `CONFIG_HOTPLUG_CPU` kernel config.

Reply:
this is interesting, I've the `CONFIG_HOTPLUG_CPU=y` config but no online file for `cpu0`

> Perhaps some cpus allow the cpu0 to be disabled and the kernel then exposes their online file too?

It could be :thinking: